controller/api/v1alpha1/prautomation_types.go

```diff
@@ -0,0 +1,318 @@
+package v1alpha1
+
+import (
+	console "github.com/pluralsh/console-client-go"
+	"github.com/pluralsh/polly/algorithms"
+	"github.com/samber/lo"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func init() {
+	SchemeBuilder.Register(&PrAutomation{}, &PrAutomationList{})
+}
+
+// PrAutomationBindings ...
+type PrAutomationBindings struct {
+	// Create bindings.
+	// +kubebuilder:validation:Optional
+	Create []Binding `json:"create,omitempty"`
+
+	// Write bindings.
+	// +kubebuilder:validation:Optional
+	Write []Binding `json:"write,omitempty"`
+}
+
+// PrAutomationList ...
+// +kubebuilder:object:root=true
+type PrAutomationList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+
+	Items []PrAutomation `json:"items"`
+}
+
+// PrAutomation ...
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:scope=Cluster
+// +kubebuilder:subresource:status
+// +kubebuilder:printcolumn:name="ID",type="string",JSONPath=".status.id",description="ID of the pr automation in the Console API."
+type PrAutomation struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec ...
+	// +kubebuilder:validation:Required
+	Spec PrAutomationSpec `json:"spec"`
+
+	// Status ...
+	// +kubebuilder:validation:Optional
+	Status Status `json:"status,omitempty"`
+}
+
+// ConsoleID implements PluralResource interface
+func (in *PrAutomation) ConsoleID() *string {
+	return in.Status.ID
+}
+
+// ConsoleName implements PluralResource interface
+func (in *PrAutomation) ConsoleName() string {
+	if in.Spec.Name != nil && len(*in.Spec.Name) > 0 {
+		return *in.Spec.Name
+	}
+
+	return in.Name
+}
+
+func (in *PrAutomation) Attributes(clusterID *string, serviceID *string, connectionID *string) *console.PrAutomationAttributes {
+	attrs := console.PrAutomationAttributes{
+		Name:          lo.ToPtr(in.ConsoleName()),
+		Identifier:    in.Spec.Identifier,
+		Documentation: in.Spec.Documentation,
+		Title:         in.Spec.Title,
+		Message:       in.Spec.Message,
+		Branch:        in.Spec.Branch,
+		Updates:       in.Spec.Updates.Attributes(),
```

Comment:
in particular here doesn't pass the creates spec

Reply:
oh, I must have missed it